src/dynamic_contract.rs

```diff
@@ -0,0 +1,277 @@
+        assert_eq!(
+            "target/debug/deps/libfoo.dylib",
+            create_library_path("foo").to_str().unwrap()
+        );
+    }
+
+    #[test]
+    fn test_contract_id() {
+        assert_eq!(&Pubkey::new(&DYNAMIC_CONTRACT_ID), &DynamicContract::id());
+        assert_eq!(
+            true,
+            DynamicContract::check_id(&Pubkey::new(&DYNAMIC_CONTRACT_ID))
+        );
+    }
+
+    #[test]
+    fn test_contract_balance() {
+        let accounts = [
+            Account {
+                tokens: 100,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+            Account {
+                tokens: 1,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+        ];
+
+        assert_eq!(100, DynamicContract::get_balance(&accounts[0]));
+        assert_eq!(1, DynamicContract::get_balance(&accounts[1]));
+    }
+
+    #[test]
+    fn test_contract_noop() {
+        let userdata = DynamicContract::Native {
+            name: "noop".to_string(),
+            userdata: vec![0, 1],
+        };
+        let tx = Transaction {
+            signature: Signature::default(),
+            keys: vec![Pubkey::default(); 2],
+            contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            last_id: Hash::default(),
+            fee: 42,
+            userdata: serialize(&userdata).unwrap(),
+        };
+        let mut accounts = [
+            Account {
+                tokens: 100,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+            Account {
+                tokens: 1,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+        ];
+
+        DynamicContract::process_transaction(&tx, &mut accounts);
+    }
+
+    #[test]
+    #[ignore]
+    fn test_contract_print() {
+        let userdata = DynamicContract::Native {
+            name: "print".to_string(),
+            userdata: vec![0, 1],
+        };
+        let tx = Transaction {
+            signature: Signature::default(),
+            keys: vec![Pubkey::default(); 2],
+            contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            last_id: Hash::default(),
+            fee: 42,
+            userdata: serialize(&userdata).unwrap(),
+        };
+        let mut accounts = [
+            Account {
+                tokens: 100,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+            Account {
+                tokens: 1,
+                userdata: vec![0, 8],
+                contract_id: Pubkey::new(&DYNAMIC_CONTRACT_ID),
+            },
+        ];
+
+        DynamicContract::process_transaction(&tx, &mut accounts);
+    }
+
+    #[test]
+    fn test_contract_move_funds_success() {
```

Comment:
Looks like automation failed on this test (rest pass).  I didn't see any failures locally, any ideas?

test dynamic_contract::tests::test_contract_move_funds_success ... error: process didn't exit successfully: `/solana/target/debug/deps/solana-2161c8890c108436` (signal: 11, SIGSEGV: invalid memory reference)
🚨 Error: The command exited with status 101